### PR TITLE
feat: dropdown in form

### PR DESCRIPTION
**Pull request by psankhe28**
## Description

#338 

Fixes #(issue)

## Checklist

- [x] My code follows the [contributing guidelines][contributing] of this project.
- [x] I am aware that all my commits will be squashed into a single commit, using the PR title as the commit message.
- [x] I have performed a self-review of my own code.
- [ ] I have commented my code in hard-to-understand areas.
- [ ] I have updated the user-facing documentation to describe any new or changed behavior.
- [x] My changes generate no new warnings.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [ ] I have not reduced the existing code coverage.

## Comments

[contributing]: CONTRIBUTING.md

## Summary by Sourcery

This pull request introduces a new dropdown field to the form component, enhancing the form's functionality by allowing users to select from predefined options. The form component's rendering logic has been updated to support this new field type.

- **New Features**:
    - Added a dropdown (select) field to the form component, allowing users to select options from a predefined list.
- **Enhancements**:
    - Updated the form component to support the new dropdown field type, including rendering logic and field options.

### CODE REVIEW

File: packages/ecc-utils-design/src/components/form/form.ts [hunk 1]
```diff
@@ -30,14 +30,16 @@ export interface Field {
...
+    | "group"
+    | "select";
   fieldOptions?: {
     required?: boolean;
     default?: string | boolean;
     multiple?: boolean;
     accept?: string;
     returnIfEmpty?: string;
     tooltip?: string;
+    options?: Array<{ label: string; value: string }>;
```
Comment (SalihuDickson) [inline packages/ecc-utils-design/src/components/form/form.ts:42]: i dont think the `fieldOptions` object is the right place for this, as this object is for general purpose options that apply to pretty much all field types. Perhaps you can create a new object called `selectOptions` for these options.

File: packages/ecc-utils-design/src/components/form/form.ts [hunk 2]
```diff
@@ -163,15 +248,56 @@ export default class EccUtilsDesignForm extends LitElement {
...
+                  ${option.label}
+                </sl-option>
+              `
+            )}
+          </sl-select>
+        </div>
+      `;
+    }
+  
     if (!_.get(this.form, path)) {
```
Comment (SalihuDickson) [inline packages/ecc-utils-design/src/components/form/form.ts:293]: this if statement sets the default text or option for the input fields on the form, the code rendering the select field should be below it.

**Review by SalihuDickson — CHANGES_REQUESTED** [review]
great work @psankhe28, just a few minor corrections to take a look act.

File: packages/ecc-utils-design/src/components/form/form.ts [hunk 3]
```diff
@@ -121,14 +123,97 @@ export default class EccUtilsDesignForm extends LitElement {
     `;
   }
 
+  // renderInputTemplate(field: Field, path: string): TemplateResult {
```
[the remark this answers is not in the document]
Comment (psankhe28) [reply packages/ecc-utils-design/src/components/form/form.ts:126]: By mistake. Will resolve it

File: packages/ecc-utils-design/src/components/form/form.ts (hunk 2 above)
Comment (psankhe28) [reply packages/ecc-utils-design/src/components/form/form.ts:293]: The code rendering is below the if statement.

Comment (SalihuDickson) [reply packages/ecc-utils-design/src/components/form/form.ts:210]: it's not. Perhaps there is some confusion on which statement I am talking about. The statement I am talking about is this one: 
```
if (!_.get(this.form, path)) {
      if (field.fieldOptions?.default && !this.hasUpdated) {
        _.set(this.form, path, field.fieldOptions.default);
      } else if (field.fieldOptions?.returnIfEmpty) {
        _.set(this.form, path, '');
      }
    }
```

The code rendering the `select` field should be below that statement

**Review by SalihuDickson — CHANGES_REQUESTED** [review]

**Review by SalihuDickson — CHANGES_REQUESTED** [review]

**Comment by SalihuDickson** [comment]
@psankhe28, I hope you don't mind, I pushed a commit that fixed the lint issue and made sure it would be much easier to address for future contributors.

**Review by SalihuDickson — APPROVED** [review]
lgtm!

**Pull request merged**